### PR TITLE
Decrease grace period to zero to make CI testing faster

### PR DESCRIPTION
**Pull request by tomob**
Signed-off-by: Tomasz Baranski <tbaransk@redhat.com>

**What this PR does / why we need it**:

Setting the grace period of deletion of verifier pod might make CI testing faster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
any diference?

**Comment by tomob** [comment]
> any diference in the time?

Hard to say.

I've compared with other open PRs and I can't draw any conclusions. Some runs are better, some are worse. I'd need to run a more rigorous test in a controller environment.

**Comment by brybacki** [comment]
maybe now, after other fixes to verifier pod, it does not matter much

File: tests/utils/pod.go [hunk 1]
```diff
@@ -52,7 +53,7 @@ func DeletePodByName(clientSet *kubernetes.Clientset, podName, namespace string,
 	})
 	return wait.PollImmediate(2*time.Second, podDeleteTime, func() (bool, error) {
 		_, err := clientSet.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
-		if !errors.IsNotFound(err) {
+		if errors.IsNotFound(err) {
```
Comment (maya-r) [inline tests/utils/pod.go:56]: haha, oops.

**Comment by tomob** [comment]
I've made some measurements in a semi-controlled environment on my laptop. I've run the entire suite 5 times and measured run time (in seconds).

Master: runs are `7225.33, 7203.28, 7126.58, 6716.57, 6932.88`, average: `7040,878`, median: `7126.58`.
This PR: `5967.21, 6444.85, 5937.15, 6017.37, 6103.24`, average: `6093,964`, median: `6017.37`.

So it seems that this PR speeds tests up ~13% (*on my laptop*, mind you).

**Comment by brybacki** [comment]
it's simple change, if it does not brake anything then it's worth it.

czw., 28 sty 2021, 18:29 użytkownik Tomasz Barański <
notifications@github.com> napisał:

**Comment by awels** [comment]
In the PR that stopped us waiting on the ns to be deleted I originally did this, and it introduced a bunch of flakiness at the time. Since then we have fixed a lot of things with regards to verifier pods. Can we just manually trigger the tests a bunch of times to see if we have more flakes. If not I am fine with doing this.

**Comment by awels** [comment]
Can you run the tests a few more times? I am really afraid this might introduce some flakiness.

**Comment by tomob** [comment]
@awels This is a pod-related failure but I think it's not related to the change: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/1603/pull-containerized-data-importer-e2e-k8s-1.18-upg/1357335655310430208

File: tests/utils/pod.go (hunk 1 above)
Comment (maya-r) [reply tests/utils/pod.go:56]: I've split this to a separate PR #1644 so we can backport just that part easily.

**Comment by tomob** [comment]
@awels This one keeps failing the CI with seemingly unrelated reason, but still I'm a little concerned. The important bit of this PR has been already ported by @maya-r. What shall we do with this one? Keep retesting or decide that it's not worth it and discard?

**Pull request closed without merging**